src/mgmt/srv_target.c

```diff
@@ -889,8 +842,12 @@ ds_mgmt_hdlr_tgt_map_update(crt_rpc_t *rpc)
 {
 	struct mgmt_tgt_map_update_in  *in = crt_req_get(rpc);
 	struct mgmt_tgt_map_update_out *out = crt_reply_get(rpc);
+	uint32_t			version;
+	int				rc;
 
-	if (in->tm_map_version != sys_map_version)
+	rc = crt_group_version(NULL /* grp */, &version);
+	D_ASSERTF(rc == 0, "%d\n", rc);
+	if (in->tm_map_version > version)
```

Comment:
just to confirm that the MGMT_TGT_MAP_UPDATE bcast is initiated by the MS leader synchronously one-by-one?
if possibly with multiple async bcast inflight then seems above case is possible? and what will happen if return it as error by set tm_rc = 1 as below....

Reply:
MGMT_TGT_MAP_UPDATE is indeed sent by the MS leader synchronously one by one. In theory we may get old requests that have been delayed or from an old MS leader. Here we have already executed the pre_forward callback above, so in->tm_map_version should either be < version or == version. Let me add a comment.